stake-pool/js/src/schema.ts

```diff
@@ -0,0 +1,147 @@
+import {Schema, serialize, deserializeUnchecked} from 'borsh';
+import BN from 'bn.js';
+import {SOLANA_SCHEMA, Struct, Enum} from '@solana/web3.js';
+
+export class Fee extends Struct {
+  denominator: BN;
+  numerator: BN;
+}
+
+export class AccountType extends Enum {}
+
+export class AccountTypeEnum extends Struct {}
+
+export enum AccountTypeKind {
+  Uninitialized = 'Uninitialized',
+  StakePool = 'StakePool',
+  ValidatorList = 'ValidatorList',
+}
+
+export class StakePool extends Struct {
+  accountType: AccountType;
+  manager: PublicKey;
+  staker: PublicKey;
+  depositAuthority: PublicKey;
+  withdrawBumpSeed: number;
+  validatorList: PublicKey;
+  reserveStake: PublicKey;
+  poolMint: PublicKey;
+  managerFeeAccount: PublicKey;
+  totalStakeLamports: BN;
+  poolTokenSupply: BN;
+  lastUpdateEpoch: BN;
+  fee: Fee;
+}
+
+export class ValidatorList extends Struct {
+  accountType: AccountType;
+  maxValidators: number;
+  validators: [ValidatorStakeInfo];
+}
+export class ValidatorStakeInfo extends Struct {
+  status: StakeStatus;
+  voteAccountAddress: PublicKey;
+  stakeLamports: BN;
+  lastUpdateEpoch: BN;
+}
+export class StakeStatus extends Enum {}
+
+export class StakeStatusEnum extends Struct {}
+
+export enum StakeStatusKind {
+  Active = 'Active',
+  DeactivatingTransient = 'DeactivatingTransient',
+  ReadyForRemoval = 'ReadyForRemoval',
+}
+
+export class PublicKey extends Struct {
+  value: BN;
+}
+
+export const SCHEMA: Schema = constructStakePoolSchema(SOLANA_SCHEMA);
```

Comment:
nit: This is a bit of a strange function, since it's modifying `SOLANA_SCHEMA` in place and then also returning it.  I would say the function should either clone the map, then modify and return, OR just modify it in place.

Take it or leave it, but to me the best bet would be to rename the function `addStakePoolSchema` and modify SOLANA_SCHEMA in place.

Reply:
Makes sense: have made the changes in https://github.com/solana-labs/solana-program-library/pull/1656/commits/3e8073bf1f2a560d5a746175cef3c4c1d21510a9